Buildings/Experimental/DHC/Loads/Cooling/BaseClasses/BuildingTimeSeries.mo

```diff
@@ -0,0 +1,295 @@
+    "Time constant of integrator block";
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput QReqHotWat_flow(
+    final unit="W") if have_hotWat
+    "SHW load" annotation (Placement(
+      transformation(extent={{300,-140},{340,-100}}), iconTransformation(
+      extent={{-40,-40},{40,40}},
+      rotation=-90,
+      origin={280,-340})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput QReqHea_flow(
+    final quantity="HeatFlowRate",
+    final unit="W") if have_heaLoa
+    "Heating load"
+    annotation (Placement(transformation(extent={{300,20},{340,60}}),
+      iconTransformation(extent={{-40,-40},{40,40}},rotation=-90,origin={200,-340})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput QReqCoo_flow(
+    final quantity="HeatFlowRate",
+    final unit="W") if have_cooLoa
+    "Cooling load"
+    annotation (Placement(transformation(extent={{300,-20},{340,20}}),
+      iconTransformation(extent={{-40,-40},{40,40}},rotation=-90,origin={240,-340})));
+  Modelica.Blocks.Sources.CombiTimeTable loa(
+    tableOnFile=true,
+    tableName="tab1",
+    fileName=Modelica.Utilities.Files.loadResource(
+      filNam),
+    extrapolation=Modelica.Blocks.Types.Extrapolation.Periodic,
+    y(each unit="W"),
+    offset={0},
+    columns={2},
+    smoothness=Modelica.Blocks.Types.Smoothness.MonotoneContinuousDerivative1)
+    "Reader for thermal loads (y[1] is cooling load, y[2] is heating load)"
+    annotation (Placement(transformation(extent={{-280,-10},{-260,10}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant minTSet(
+    k=293.15,
+    y(final unit="K",
+      displayUnit="degC"))
+    "Minimum temperature set point"
+    annotation (Placement(transformation(extent={{-280,170},{-260,190}})));
+  Buildings.Controls.OBC.CDL.Continuous.Sources.Constant maxTSet(
+    k=297.15,
+    y(final unit="K",
+      displayUnit="degC"))
+    "Maximum temperature set point"
+    annotation (Placement(transformation(extent={{-280,210},{-260,230}})));
+  replaceable Buildings.Experimental.DHC.Loads.BaseClasses.Validation.BaseClasses.FanCoil2PipeHeating terUniHea(
+    final k=k,
+    final Ti=Ti) if have_heaWat
+  constrainedby Buildings.Experimental.DHC.Loads.BaseClasses.PartialTerminalUnit(
+    redeclare final package Medium1=Medium,
+    redeclare final package Medium2=Medium2,
+    final allowFlowReversal=allowFlowReversal,
+    final facMul=facMulHea,
+    final facMulZon=1,
+    final QHea_flow_nominal=QHea_flow_nominal/facMulHea,
+    final mLoaHea_flow_nominal=mLoaHea_flow_nominal,
+    final T_aHeaWat_nominal=T_aHeaWat_nominal,
+    final T_bHeaWat_nominal=T_bHeaWat_nominal,
+    final T_aLoaHea_nominal=T_aLoaHea_nominal)
+    "Heating terminal unit"
+    annotation (Placement(transformation(extent={{70,-22},{90,-2}})));
+  Buildings.Experimental.DHC.Loads.BaseClasses.FlowDistribution disFloHea(
+    redeclare final package Medium=Medium,
+    final allowFlowReversal=allowFlowReversal,
+    m_flow_nominal=mHeaWat_flow_nominal,
```

Comment:
check if `final` is needed?

Reply:
Thank you for bringing this up, I believe it is needed to add `final` prefix because its value can be propagated by `mHeaWat_flow_nominal`.